tools/build/Library/InstallUnpacker/classes/Download.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+class PrestashopCouldNotDownloadLatestVersionException extends \Exception
+{
+}
+
+/**
+ * In charge of downloading the latest Prestashop Version
+ *
+ * Most methods are copied from https://github.com/PrestaShop/autoupgrade/blob/master/classes/TaskRunner/Upgrade/Download.php
+ * and https://github.com/PrestaShop/autoupgrade/blob/master/classes/Tools14.php
+ */
+class Download
+{
+    // @todo: what happens for PS 1.8+ ?
+    const ADDONS_API_RELEASES_XML_FEED = 'https://api.prestashop.com/xml/channel17.xml';
```

Comment:
I can answer this question.

Actually, we shouldn't have this file available. We are supposed to only have `https://api.prestashop.com/xml/channel.xml` and have all the existing versions in it. `channel17.xml` has been created for a specific version of 1-click upgrade, only upgrading to PrestaShop 1.7.

You should have channel.xml here.

Reply:
@matks, PrestaShop 1.7 details is now available in channel.xml, can you please replace the file name?